orbit/pkg/table/extension.go

```diff
@@ -85,28 +86,53 @@ func (r *Runner) Interrupt(err error) {
 	}
 }
 
-// waitForSocket waits for the osquery socket to exist.
+// waitExtensionSocket waits until the osquery extension manager socket is ready.
 //
-// This method was copied from osquery-go because we can't rely on option.ServerTimeout.
-// Such timeout is used both for waiting for the socket and for the thrift transport.
-func waitForSocket(sockPath string, timeout time.Duration) error {
+// We can't rely on osquery-go's option.ServerTimeout. Such timeout is used both for waiting
+// for the socket and for the thrift transport.
+func waitExtensionSocket(sockPath string, timeout time.Duration) error {
 	ticker := time.NewTicker(200 * time.Millisecond)
 	defer ticker.Stop()
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
+FOR_LOOP:
 	for {
 		select {
 		case <-ctx.Done():
-			return ctx.Err()
+			return errors.New("extension socket stat timeout")
 		case <-ticker.C:
 			switch _, err := os.Stat(sockPath); {
 			case err == nil:
-				return nil
+				break FOR_LOOP
```

Comment:
Do you think it would be more clear if the bottom section was refactored into a function like `waitExtensionServerResponsive(sockPath, ctx)`? I'm not used to seeing the `break FOR_LOOP` syntax, but I'm not sure that my idea is more clear.

Reply:
FWIW, we use labels in a few other places, eg: http://github.com/fleetdm/fleet/blob/3c9e1436deef9ffa7d9d37e51a2420cc1ed97f49/server/service/live_queries.go#L115-L115